### ISSUE TITLE
Add support for "explicit content" beatmap markers & warnings

### ISSUE THREAD

**Issue by frenzibyte** [ppy/osu#11460]
Currently going through this one now that it's live web-side, can't assign myself (no perms) so would like somebody to do so.

Also submitting this for feedback/suggestions over how handling should go when attempting to join a room that has an *explicit* beatmap selected at the moment, or when attempting to spectate a player that is playing an *explicit* beatmap.

Thinking about going simple and show a warning similar to the beatmap viewing one (show once, disable warning, go back) when attempting to do any of the above operations. 

And maybe, if necessary, adding a checkbox to hide any certain room/player that is involved with an *explicit* beatmap. (can just use the user's configuration towards explicit content for hiding as well), what do you think?

**Comment by bdach** [comment]
I think the scope of this issue might be reaching a bit far to be honest. This is definitely not something that will be resolved by a single pull.

**Comment by frenzibyte** [comment]
Yeah, I submitted this for suggestions on the above and to mark out that I've been working on this. (for the record, I'm finished with beatmap listing and overlay stuff, will PR them out tomorrow)

**Comment by Gamingzone123** [comment]
Should probably have a setting to ignore the explicit content warning, a bit like how you can set listings to show explicit beatmaps regardless, on that note should probably have a setting for the epilepsy warning too, but that's another issue

**Comment by bdach** [comment]
@Gamingzone123 did you perhaps miss this?

![2022-02-21-224846_521x465_scrot](https://user-images.githubusercontent.com/20418176/155030086-71e1c5fc-10e0-401d-8a8d-c8d98004ffc8.png)

**Comment by Gamingzone123** [comment]
I was talking about in-game when starting the beatmap

**Comment by mk56-spn** [comment]
seems kinda pointless since the NSFW backgrounds show in song select

**Comment by Userlog2** [comment]
> seems kinda pointless since the NSFW backgrounds show in song select

Almost all "Explicit" marks refer to Explicit lyrics. 
An idea could be that Explicit songs are muted in song select unless the user changes a setting to play them automatically like other songs.

**Comment by SirHonkkk** [comment]
a feture to disable explicit beatmaps in multi would be awsomesauce.
